app/src/main/java/com/nft/signdoc/ui/auth/SignupActivity.kt

```diff
@@ -20,15 +41,140 @@ class SignupActivity : BaseActivity() {
 
     private val binding by viewBinding(ActivitySignupBinding::inflate)
 
+    private val userViewModel : UserViewModel by viewModels()
+
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(binding.root)
+        when (userViewModel.loginType) {
+            "phone" -> binding.accountId.setText(userViewModel.currentEmail.split("@")[0].replace(".", ""), TextView.BufferType.EDITABLE)
+            "email" -> binding.accountId.setText(userViewModel.currentPhone.replace("+", ""), TextView.BufferType.EDITABLE)
+            else -> {
+
+            }
+        }
+        initViews()
         listenToViewEvents()
     }
 
+    private fun initViews(){
+        binding.termsText.makeLinks(Pair("Terms & Conditions", View.OnClickListener {
+            val browserIntent = Intent(Intent.ACTION_VIEW)
+            browserIntent.data = Uri.parse("https://terms.nftmakerapp.io/")
+            this.startActivity(browserIntent)
+        }), Pair("Privacy Policy", View.OnClickListener {
```

Comment:
Same code is also there in createAccActivity file. Why not  create a global function in Util class and just call that function from both places.

Reply:
am taking this into consideration and repushing